staging/src/k8s.io/controller-manager/pkg/features/kube_features.go

```diff
@@ -32,19 +32,26 @@ const (
 	// of code conflicts because changes are more likely to be scattered
 	// across the file.
 
+	// owner: @nckturner
+	// kep:  http://kep.k8s.io/2699
+	// alpha: v1.27
+	// Enable webhook in cloud controller manager
+	CloudControllerManagerWebhook featuregate.Feature = "CloudControllerManagerWebhook"
+
+	// owner: @danwinship
+	// alpha: v1.27
+	//
+	// Enables dual-stack values in the
+	// `alpha.kubernetes.io/provided-node-ip` annotation
+	CloudDualStackNodeIPs featuregate.Feature = "CloudDualStackNodeIPs"
+
 	// owner: @alexanderConstantinescu
 	// kep: http://kep.k8s.io/3458
 	// beta: v1.27
 	//
 	// Enables less load balancer re-configurations by the service controller
 	// (KCCM) as an effect of changing node state.
 	StableLoadBalancerNodeSet featuregate.Feature = "StableLoadBalancerNodeSet"
-
-	// owner: @nckturner
-	// kep:  http://kep.k8s.io/2699
-	// alpha: v1.27
-	// Enable webhook in cloud controller manager
-	CloudControllerManagerWebhook featuregate.Feature = "CloudControllerManagerWebhook"
 )
 
 func SetupCurrentKubernetesSpecificFeatureGates(featuregates featuregate.MutableFeatureGate) error {
```

Comment:
I didn't know we'd grown a new feature gate package.

Is this method called from `k8s.io/kubernetes/pkg/features#init` so all the features defined here are available in the global kubernetes featuregate state holder?

Reply:
looks like @deads2k was around when this went in - https://github.com/kubernetes/kubernetes/pull/92201#discussion_r497932809 - would be good to get some guidance on how this was intended to work written down, and see if we're still happy with this approach. Seems twisty to me, but I'm not really sure what the intent here was.

Reply:
From a newly-aware-of-this-mess POV, I'd advocate for a single staging repo that defines ALL of the gates use in ALL core components.  Anything else is asking too much from reviewers and contributors to understand.